packages/world-modules/src/modules/erc721-puppet/ERC721System.sol

```diff
@@ -64,7 +64,7 @@ contract ERC721System is IERC721Mintable, System, PuppetMaster {
 
     string memory baseURI = _baseURI();
     string memory _tokenURI = TokenURI.get(_tokenUriTableId(_namespace()), tokenId);
-    _tokenURI = bytes(_tokenURI).length > 0 ? _tokenURI : string(abi.encodePacked(tokenId));
+    _tokenURI = bytes(_tokenURI).length > 0 ? _tokenURI : uintToString(tokenId);
```

Comment:
Agreeing with @alvrs, we can lift the [`Strings`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Strings.sol) library that OZ uses [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol#L92) in their ERC721 implementation.

In MUD we straight up copy a lot of files: https://github.com/latticexyz/mud/blob/main/packages/world/src/ERC165Checker.sol